sdk/core/core-lro/src/poller.ts

```diff
@@ -21,7 +21,20 @@ export class PollerCancelledError extends Error {
   }
 }
 
-export abstract class Poller<TState, TResult> {
+export interface PollerInterface<TState, TResult> {
```

Comment:
I'd prefer to either:

1. call this interface `PollerLike`
2. call this interface `Poller` and the class below `BasePoller`

Reply:
Let's do PollerLike